discovery_engine_core/core/src/stack/ops/breaking.rs

```diff
@@ -43,11 +42,22 @@ impl Ops for BreakingNews {
     }
 
     fn configure(&mut self, config: &EndpointConfig) {
+        self.client = Client::new(config.api_key.clone(), config.api_base_url.clone());
         self.markets.replace(Arc::clone(&config.markets));
     }
 
     async fn new_items(&self, _key_phrases: &[KeyPhrase]) -> Result<Vec<Article>, GenericError> {
-        Ok(vec![])
+        Ok(if let Some(markets) = self.markets.as_ref() {
+            let mut articles = Vec::new();
+            let page_size = Some(20); // TODO pass through config later
+            for market in markets.read().await.clone() {
+                let query = HeadlinesQuery { market, page_size };
+                articles.extend(self.client.headlines(&query).await?);
```

Comment:
This will terminate with an error even if only one request returned an error. 
We don't have a way at the moment to signal only one error, but is better for to return an error only when all of them return an error

Reply:
ok, adjusted so that errors are now recorded in a vec. in the case where no articles were retrieved, only errors, (for now) the last error is returned.